docs/DeveloperGuide.md

```diff
@@ -1,12 +1,220 @@
+Parking lot data)
+   - This method makes all 5 request asynchronously and receives in sequence.
+   - The method will concatenate the data together for data processing by other parts of the program.
+   - The processed data will then be stored in a local data file.
+   
+The following sequence diagram shows how data is fetched using the `update` command.
+![Update Data API](images/UpdateApiSequenceDiagram.png)
+
+
+The API component is also able to:
+- Loads in key from a local file storage (in txt format).
+- Authenticate user API key. If no user key inputted, default key will be loaded.
+- Get API authentication status.
+
+The following sequence diagram shows how the API key is loaded.
+![Api Loading Sequnce Diagram](images/LoadApiSequenceDiagram.png)
+
+
+#### Storage Component
+
+#### Implementation
+![Sequence Diagram](images/LoadFileSequenceDiagram.png)
+##### FileWriter
+
+##### FileLoader
+
+#### UI Component
+
+The user-facing parts of the program are implemented with the Ui class.
+
+It implements the following main functions:
+
+- print() - Prints a string to System.out.
+- getLine() - Asks user for input and returns input.
+- getCommand() - Asks user for a command input and returns input.
+- showLogo() - Prints ASCII art of a car and parKING logo.
+- printResult() - Shows the result of a command execution to the user. Includes additional formatting of the results of 
+different commands.
+- printError() - Print exception message.
+- changeScanner() - Changes the scanner for the Ui object. To be used for JUnit testing.
+- getSeparatorString() - Returns a separator string.
+
+#### Logic Component
+How the parsing works:
+
+* When called upon to parse a user command, the `Parser` class creates an `ABCCommandParser` (`ABC` is a placeholder 
+for the specific command name eg. `FindCommandParser`) which uses the other classes to parse the user command and
+create an `ABCCommand` object (eg. `FindCommand`), which the `Parser` returns back as a `Command` object.
+
+* All `ABCCommandParser` classes (eg. `FindCommandParser`, ``SearchCommandParser` etc) inherit from the `Parser` 
+interface so that they can be treated similarly where possible.
+
+How the command works:
+
+* When the `Parser` returns back as a `Command` object, it will run `execute`, which is a method to execute the command
+and return the intended result. `Execute` returns a `CommandResult`, which is an object that takes in `String` and
+`CarparkList` and returns what the user will be seeing on the terminal.
+* Each command will have its own class `ABCCommand` (`ABC` is a placeholder for the specific command name eg. 
+`FindCommand`) as an extension of the `Command` class.
+* All `ABCCommand` classes will have an override `execute` method, which will override the `execute` method in `Command` 
+and return the respective `CommandResult` result of the Command.
+
+![Logic Class Diagram](images/LogicClassDiagram.png)
```

Comment:
Same case here, do double check that you use the correct visibilty indicators as specified by the module.